### PR TITLE
feat(profile, edit-profile): add db connection, some enhancements

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- Provided db connection for profile and edit-profile pages, data sync between components and db,  added some enhancements

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://ticket-url)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/types/user.ts [hunk 1]
```diff
@@ -104,20 +116,47 @@ export type IUserItem = {
...
+  // firstName: string;
+  // lastName: string;
+  // role: string;
+  // email: string;
+  // country: string;
+  // city: string;
+  // agency?: string;
+  // avatarUrl?: string;
+  // phone: string;
+  // description: string;
```
Comment (canterbery) [inline src/types/user.ts]: Commented code

File: src/sections/user/view/user-profile-view.tsx [hunk 2]
```diff
@@ -24,19 +28,42 @@ const TABS = [
...
+        const res = await getUserById({ id: userId });
 
-  const handleChangeTab = useCallback((event: React.SyntheticEvent, newValue: string) => {
-    setCurrentTab(newValue);
-  }, []);
+        if ('data' in res) {
+          setData(res.data);
+        }
+      } catch (error) {
+        console.error('Error fetching user data', error);
```
Comment (OlgaSavina) [inline src/sections/user/view/user-profile-view.tsx]: Console

File: src/types/user.ts (hunk 1 above)
Comment (anette1983) [reply src/types/user.ts]: fixed

File: src/sections/user/view/user-profile-view.tsx (hunk 2 above)
Comment (anette1983) [reply src/sections/user/view/user-profile-view.tsx]: fixed

**Review by AKutusenko — APPROVED** [review]

**Pull request merged**